query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -669,9 +899,15 @@ case class ApplyLimitFunction(vectors: PeriodicSeriesPlan,
   override def replacePeriodicSeriesFilters(filters: Seq[ColumnFilter]): PeriodicSeriesPlan =
     this.copy(columnFilters = LogicalPlan.overrideColumnFilters(columnFilters, filters),
               vectors = vectors.replacePeriodicSeriesFilters(filters))
+
+  override def useHigherLevelAggregatedMetricIfApplicable(params: HierarchicalQueryExperience,
```

Comment:
I find in the implementation child plans have to know their parents.
Personally I feel it is a it strange because it is like the children make the decision. 
Just curious if there would be an alternative?

Reply:
The reason for this is because the column filters that we need to update, is stored only in the raw series. The plans which stores raw series is at leaf level. Hence, we need to recurse all the way to the leaf and update the plan at that level.